awx/main/tasks.py

```diff
@@ -790,8 +790,8 @@ def run(self, pk, isolated_host=None, **kwargs):
             safe_env = self.build_safe_env(env, **kwargs)
 
             # handle custom injectors specified on the CredentialType
-            if hasattr(instance, 'all_credentials'):
-                credentials = instance.all_credentials
+            if hasattr(instance, 'credentials'):
+                credentials = instance.credentials.all()
             elif hasattr(instance, 'credential'):
                 credentials = [instance.credential]
```

Comment:
@ryanpetrello this might be a hazard of promoting `credentials` from the JT to the UJT model. If it detects `credentials`, then it won't try to detect the `credential` field.

The predictable solution would be to change this check to

```python
             if hasattr(instance, 'credential'):
                  credentials += [instance.credential]
````

But you can't do this so simply because `credential` as a non-field property of jobs. Maybe you could check if `instance.credential` is an integer, and if so, don't append it.

Reply:
Or you could leverage the `all_credentials` property for all unified job models. There's too much logic going on in tasks.py with this, I'd like to see the custom processing of the credentials list moved out somewhere else.

Reply:
Yea, this is tricky because it sort of relies on our model being somewhat in flight in terms of migrating credentials for non-JT/Job.  I'm going to brainstorm and experiment a bit.

Reply:
Okay, I've got something uploaded that address this.

Long-term - especially after you continue on your work with https://github.com/ansible/awx/issues/169 - I'd love for all `.credential` foreign keys to go away and completely be replaced `UJT.credentials`.   If/when we do that, we'll need to probably work out some additional validation for certain types of jobs (e.g., "only one Machine/SSH for adhoc").